backend/src/appointment/database/repo/slot.py

```diff
@@ -114,3 +114,21 @@ def delete(db: Session, slot_id: int):
     db.delete(db_slot)
     db.commit()
     return db_slot
+
+
+def reject(db: Session, slot_id: int) -> models.Slot | None:
+    """update booking status for slot of given id as declined"""
+    db_slot = get(db, slot_id)
+    db_slot.booking_status = models.BookingStatus.declined
+    db.commit()
+    db.refresh(db_slot)
+    return db_slot
+
+
+def cancel(db: Session, slot_id: int) -> models.Slot | None:
+    """update booking status for slot of given id as cancelled"""
+    db_slot = get(db, slot_id)
+    db_slot.booking_status = models.BookingStatus.cancelled
+    db.commit()
+    db.refresh(db_slot)
+    return db_slot
```

Comment:
This is fine, I was just thinking if we want to unify those into a `set_status(db: Session, slot_id: int, status: models.BookingStatus)` function and call this instead of duplicating the save logic for each status.

Reply:
Valid point. Although normally we used to keep the repository functions rather atomic and put the business logic in the controllers. So any misuse of a possible set_status function should be handled/validated before calling it.

But I'm also open for both!